pkg/aws/ec2/pricing_map.go

```diff
@@ -326,3 +390,46 @@ func ListSpotPrices(ctx context.Context, client ec2client.EC2) ([]ec2Types.SpotP
 	}
 	return spotPrices, nil
 }
+
+func ListStoragePrices(ctx context.Context, region string, client pricingClient.Pricing) ([]string, error) {
+	input := &pricing.GetProductsInput{
+		ServiceCode: aws.String("AmazonEC2"),
+		Filters: []types.Filter{
+			{
+				Field: aws.String("regionCode"),
+				Type:  "TERM_MATCH",
+				Value: aws.String(region),
+			},
+			// Get prices for EBS Volumes
+			{
+				Field: aws.String("productFamily"),
+				Type:  "TERM_MATCH",
+				Value: aws.String("Storage"),
+			},
+		},
+	}
+
+	return getPricesFromProductList(ctx, input, client)
+}
+
+func getPricesFromProductList(ctx context.Context, input *pricing.GetProductsInput, client pricingClient.Pricing) ([]string, error) {
+	var productOutputs []string
+
+	for {
+		products, err := client.GetProducts(ctx, input)
+		if err != nil {
+			return productOutputs, err
+		}
+
+		if products == nil {
+			break
+		}
+
+		productOutputs = append(productOutputs, products.PriceList...)
+		if products.NextToken == nil {
+			break
+		}
+		input.NextToken = products.NextToken
```

Comment:
I haven't directly unit tested `getPricesFromProductsList`, because we get that for free from having both `ListStoragePrices` and `ListOnDemandPrices` tested. I could add a unit test, but I find that it doesn't add that much value, and I'd rather test the public methods instead.

As for adding it to the interface, I only added the other ones to the EC2 interface because I needed them for the compiler to stop screaming at me and to be able to test the new functionality. In this case, these methods are tested without the need to add them to the Pricing interface, so I don't see what the value would be in having them there too 🤔